axelrod/strategies/grudger.py

```diff
@@ -239,3 +239,73 @@ def strategy(opponent: Player) -> Action:
         if opponent.defections:
             return C
         return D
+
+class GeneralSoftGrudger(Player):
+    """
+    A generalization of the SoftGrudger strategy. SoftGrudger punishes by playing:
+    D, D, D, D, C, C. after a defection by the opponent.  GeneralSoftGrudger
+    only punishes after its opponent defects a specified amount of times consecutively.
+    The punishment is in the form of a series of defections followed by a 'penance' of
+    a series of consecutive cooperations.
+    """
```

Comment:
Could you add:

```
Names:

- General Soft Grudger: ...
```

and after the name, either a reference to the literature (if one exists to your knowledge) **or** say that it's an original name, like here: http://axelrod.readthedocs.io/en/latest/reference/all_strategies.html#axelrod.strategies.ann.EvolvedANN

The idea of that is that it auto generates http://axelrod.readthedocs.io/en/latest/reference/all_strategies.html which is a place that users can `ctrl + F` strategies.